[docs] Describe video recording

VasilyStrelyaev opened the pull request:
\cc @DevExpress/testcafe-docs 

It's awfully cross-linked (even between screenshots and video stuff), so ideally I should separate a `Screenshots and Videos` topic to gather it all.

And I haven't described the new `${GENERIC_TEST_NAME}` and `${GENERIC_RUN_NAME}` path patterns yet. Their description will fit well only when we create the `Screenshots and Videos` topic. As I've discussed with @AndreyBelym , the only good reason for customers to use them is if they want a configuration file setting that would work for both enabled and disabled quarantine mode. IDK how important this is.

AndreyBelym commented on docs/articles/documentation/using-testcafe/command-line-interface.md:
```diff
@@ -356,6 +359,79 @@ testcafe all tests/sample-fixture.js -s screenshots -p "${DATE} ${TIME}/test ${T
+
+The relative path to a video file and its name are composed according to the following patterns:
+
+* `${DATE}_${TIME}/test-${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4` if the [quarantine mode](#-q---quarantine-mode) is disabled;
```
There is another pattern if you record a video in a single file:
```
 `${DATE}_${TIME}/${USERAGENT}/${FILE_INDEX}.mp4`
```
(it's the main reason why I bothered creating the `GENERIC_TEST_NAME` placeholder.

AndreyBelym commented on docs/articles/documentation/using-testcafe/command-line-interface.md:
```diff
@@ -356,6 +359,79 @@ testcafe all tests/sample-fixture.js -s screenshots -p "${DATE} ${TIME}/test ${T
+`singleFile` | Boolean | `true` to save the entire recording as a single file; `false` to create a separate file for each test. | `false`
+`ffmpegPath` | String | The path to the FFmpeg codec executable. | Auto-detected
+`pathPattern` | String | A pattern that defines how TestCafe composes the relative path to a video file and the file name. Use the same placeholders as in the [screenshot path patterns](#-p-pattern---screenshot-path-pattern-pattern). See an example below. | See [--video](#--video-basepath).
+`timeStamp` | Date | The timestamp recorded as the video creation time. | The moment when the test task starts.
```
It's not a public option, we should not list it here.

AndreyBelym commented on docs/articles/documentation/using-testcafe/command-line-interface.md:
```diff
@@ -356,6 +359,79 @@ testcafe all tests/sample-fixture.js -s screenshots -p "${DATE} ${TIME}/test ${T
+The relative path to a video file and its name are composed according to the following patterns:
+
+* `${DATE}_${TIME}/test-${TEST_INDEX}/${USERAGENT}/${FILE_INDEX}.mp4` if the [quarantine mode](#-q---quarantine-mode) is disabled;
+* `${DATE}_${TIME}/test-${TEST_INDEX}/run-${QUARANTINE_ATTEMPT}/${USERAGENT}/${FILE_INDEX}.mp4` if the [quarantine mode](#-q---quarantine-mode) is enabled.
```
We don't use a separate path pattern for quarantine mode, all quarantine attempts are recorded to a single file with the first default path pattern specified in the topic.

VasilyStrelyaev commented:
This way it looks better. FPR

VasilyStrelyaev commented:
Made the following changes:

* Removed the trailing slash as @kirovboris suggested,
* Renamed a section `Paths Where Screenshots and Videos Are Saved` -> just `Where Screenshots and Videos Are Saved`
* Restored some lost old info in the CLI topic (about `cmd.exe` specifics).

helen-dikareva commented on docs/articles/documentation/using-testcafe/common-concepts/screenshots-and-videos.md:
```diff
@@ -0,0 +1,297 @@
+---
+# Screenshots and Videos
+
+TestCafe allows you to take screenshots of the webpage under test and record videos of test runs.
```
IFAIK we have a common limitation about making screenshots and videos on remote browsers (see [here](https://devexpress.github.io/testcafe/documentation/test-api/actions/take-screenshot.html)). I think we can place it in this section and add a cross-link to the `Take Screenshot` action` section.

helen-dikareva replied:
we can also mention about limitation for video recording: we can do it only for Chrome and Firefox

VasilyStrelyaev replied:
> we can do it only for Chrome and Firefox

wow, how come I didn't know it till now?

helen-dikareva replied:
idk)

VasilyStrelyaev commented:
@testcafe-build-bot \retest

AndreyBelym commented:
They're renamed to `{TEST_ID}` and `{RUN_ID}` and described here: https://github.com/DevExpress/testcafe/pull/3375/files#diff-31ca4e6ffd60007e3918ce457181c81aR307

VasilyStrelyaev commented:
After we merged that PR https://github.com/DevExpress/testcafe/pull/3391 test will fail.
Ignore them if the're just about the config file link.

AndreyBelym commented:
Yes, the tests failed due to the missing config example in the `master` branch.
